pkg/dmesg/watcher.go

```diff
@@ -0,0 +1,134 @@
+package dmesg
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/leptonai/gpud/log"
+	"github.com/leptonai/gpud/pkg/process"
+)
+
+type LogLine struct {
+	Timestamp time.Time
+	Content   string
+	Error     string
+}
+
+type Watcher interface {
+	Watch() <-chan LogLine
+	Close()
+}
+
+func NewWatcher(cmds ...string) (Watcher, error) {
```

Comment:
this is bad. watcher should be dmesg specific. cannot just pass in whatever command

Reply:
if you want to enable test, do NewTestWatcher(). also, we should test dmesg directly... not something else. you can write to /dev/kmsg i guess or maybe not. cc @cardyok